File: src/de/hsh/amir/logic/Gegner.java
package de.hsh.amir.logic;

import common.actor.Actor;

public final class Gegner extends Actor {

    private static final double START_Y = 10;
    private static double GEGNER_SPEED = 3;

    private static final String enemyPicture = "/de/hsh/amir/resources/enemyFigur.png";

    Gegner( final double x ) {
        super( enemyPicture, x, START_Y );
    }

    Gegner( String enemyPicture, final double x ) {
        super( enemyPicture, x, START_Y );
    }

    public final void move() {
        setPos(this.getX(), this.getY() + GEGNER_SPEED);
    }

    void moveDiagonal() {
        setPos(this.getX() + GEGNER_SPEED / 4, this.getY() + GEGNER_SPEED);
    }

    /**
     * Setzt Gegnergeschwindigkeit GLOBAL!!! auf den übergebenen Wert.
     * @param speed
     */
    public static void setGegnerSpeed(double speed) {
        GEGNER_SPEED = speed;
    }
}

File: Games/Amir/src/de/hsh/amir/logic/Gegner.kt
package de.hsh.amir.logic

import common.actor.Actor

class Gegner internal constructor(x: Double) : Actor(enemyPicture, x, START_Y) {

    fun move() {
        setPos(x, y + GEGNER_SPEED)
    }

    fun moveDiagonal() {
        setPos(x + GEGNER_SPEED / 4, y + GEGNER_SPEED)
    }

    companion object {
        private const val START_Y = 10.0
        private var GEGNER_SPEED = 3.0
        private const val enemyPicture = "/de/hsh/amir/resources/enemyFigur.png"

        /**
         * Setzt Gegnergeschwindigkeit GLOBAL!!! auf den übergebenen Wert.
         * @param speed
         */
        @JvmStatic
        fun setGegnerSpeed(speed: Double) {
            GEGNER_SPEED = speed
        }
    }
}
